### ISSUE TITLE
A promise was created in a handler but was not returned from it

### ISSUE THREAD

**Issue by artursgirons** [coopernurse/node-pool#251]
This started to show up when upgraded from 3.4.0 to 3.6.0.
Using `sequelize` and this happens when calling `connection.authenticate()`

Warning: a promise was created in a handler at /workspace/node_modules/generic-pool/lib/Pool.js:319:14 but was not returned from it, see http://goo.gl/rRqMUw
    at Function.Promise.cast (/workspace/node_modules/bluebird/js/release/promise.js:196:13)

**Comment by KEVINdEsBuRgErS** [comment]
Hi, same issue but it's not just for sequelize.authenticate(), same for all promise of sequelize.
Thanks.
```
const Sequelize = require("sequelize");

const sequelize = new Sequelize("test", "root", null, {
  dialect: "mysql",
  operatorsAliases: false
});

sequelize
  .authenticate()
  .then(() => {
    console.log("[sequelize]", "Successfully authenticated");
  })
  .catch((err) => {
    console.error("[sequelize]", "Unable to connect to the database:", err.message);
  });
```

> kevin@Kevin-PC:~/dev/test$ NODE_ENV=development node sequelizetest
(node:32433) Warning: a promise was created in a handler at /dev/test/node_modules/generic-pool/lib/Pool.js:319:14 but was not returned from it, see http://goo.gl/rRqMUw
    at Function.Promise.cast (/dev/test/node_modules/bluebird/js/release/promise.js:196:13)
Executing (default): SELECT 1+1 AS result
[sequelize] Successfully authenticated

**Comment by eseliger** [comment]
I was thinking of this line: https://github.com/coopernurse/node-pool/blob/master/lib/Pool.js#L167 but its just a guess

**Comment by sandfox** [comment]
Ok, I'll see what I can work out about whats happening here

**Comment by DiegoRBaquero** [comment]
@eseliger It has its handler in line 170. But we should add a catch.

**Comment by sandfox** [comment]
Ah, I think it was most likely my removal of the `return null` in https://github.com/coopernurse/node-pool/commit/f5a93df86e58b2fd50e7f8c12c1c05c0a14b5572#diff-1b589a05fcf1509343500faf9cd70bdaL314
 
I did wonder why it was there to start with... now we know!

**Comment by sandfox** [comment]
fixed and published in v3.6.1
